From 96a2c462fdc0c1d3ddf2e59993ad5491d1331754 Mon Sep 17 00:00:00 2001
From: Jon Nettleton <jon@solid-run.com>
Date: Mon, 30 Aug 2021 13:45:42 +0200
Subject: [PATCH] [PATCH 1009/1013] hotspot: fix adlc flags
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

adlc is built using the native toolchain, not the crosscompiler. It however
was incorrectly using flags meant for the crosscompiler.

*edit*
Make was failing to remove all the flags, and really it is unnecessary. Just
bring in the build flags and remove SYSDEFS which also had cross compile
specifics in iit.

Upstream-Status: Inappropriate [Yocto-specific fixes]

Signed-off-by: Erkka Kääriä <erkka.kaaria@intel.com>
Signed-off-by: Richard Leitner <richard.leitner@skidata.com>
---
 hotspot/make/linux/makefiles/adlc.make | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hotspot/make/linux/makefiles/adlc.make b/hotspot/make/linux/makefiles/adlc.make
index 92b8b9b3..447d7c82 100644
--- a/hotspot/make/linux/makefiles/adlc.make
+++ b/hotspot/make/linux/makefiles/adlc.make
@@ -28,6 +28,10 @@
 
 include $(GAMMADIR)/make/$(Platform_os_family)/makefiles/rules.make
 
+# Yocto specific fix - only use build machine flags using hosttools
+CXXFLAGS := $(BUILD_CXXFLAGS)
+CFLAGS := $(BUILD_CFLAGS)
+
 # #########################################################################
 
 # OUTDIR must be the same as AD_Dir = $(GENERATED)/adfiles in top.make:
@@ -59,7 +63,7 @@ Src_Dirs_I += $(GAMMADIR)/src/share/vm/adlc $(GENERATED)
 INCLUDES += $(Src_Dirs_I:%=-I%)
 
 # set flags for adlc compilation
-CXXFLAGS = $(SYSDEFS) $(INCLUDES)
+CXXFLAGS = $(INCLUDES)
 
 # Force assertions on.
 CXXFLAGS += -DASSERT
-- 
2.27.0

